rasterio/_base.pyx

```diff
@@ -109,26 +110,31 @@ def driver_can_create_copy(drivername):
 cdef class DatasetBase(object):
     """Dataset base class."""
 
-    def __init__(self, path=None, options=None):
-        cdef GDALDriverH driver = NULL
+    def __init__(self, path=None, driver=None, **kwargs):
         cdef GDALDatasetH hds = NULL
-        cdef const char *path_c = NULL
+        cdef int flags = 0
 
         self._hds = NULL
 
         if path is not None:
-            path_b = vsi_path(*parse_path(path)).encode('utf-8')
-            path_c = path_b
+            filename = vsi_path(*parse_path(path))
+
+            # driver may be a string or list of strings. If the
```

Comment:
The option for `drivers` to be a list of drivers for `rasterio.open(mode='r')` is undocumented in `rasterio.open()`.

Reply:
I think passing a list of drivers is rarely useful, so I didn't want to emphasize it too much in the docs.